security-admin/src/main/java/org/apache/ranger/security/web/filter/AltiscaleRangerAuthFilter.java

```diff
@@ -0,0 +1,477 @@
+		params.put(RANGER_COOKIE_DOMAIN.replaceFirst(CONFIG_PREFIX+"alt-kerberos.",""), PropertiesUtil.getProperty(RANGER_COOKIE_DOMAIN));
+		params.put(RANGER_NON_BROWSER_USER_AGENTS.replaceFirst(CONFIG_PREFIX,""), PropertiesUtil.getProperty(RANGER_NON_BROWSER_USER_AGENTS));
+		try {
+			params.put("kerberos.principal", SecureClientLogin.getPrincipal(PropertiesUtil.getProperty(PRINCIPAL,""), PropertiesUtil.getProperty(HOST_NAME)));
+		} catch (IOException ignored) {
+			// do nothing
+		}
+		params.put("kerberos.keytab", PropertiesUtil.getProperty(KEYTAB,""));
+		params.put(AUTH_TYPE, "com.altiscale.hadoop.security.AltiscaleAuthenticationHandler");
+		FilterConfig myConf = new FilterConfig() {
+			@Override
+			public ServletContext getServletContext() {
+				if (globalConf != null) {
+					return globalConf.getServletContext();
+				} else {
+					return noContext;
+				}
+			}
+
+			@SuppressWarnings("unchecked")
+			@Override
+			public Enumeration<String> getInitParameterNames() {
+				return new IteratorEnumeration(params.keySet().iterator());
+			}
+
+			@Override
+			public String getInitParameter(String param) {
+				return params.get(param);
+			}
+
+			@Override
+			public String getFilterName() {
+				return "KerberosFilter";
+			}
+		};
+		super.init(myConf);
+		LOG.debug("AltiscaleRangerAuthFilter.init(FilterConfig filterConfig) ==> ended");
+	}
+
+	/**
+	 * This function performs post tasks after {@link AuthenticationFilter#doFilter(ServletRequest, ServletResponse, FilterChain)}
+	 * to create a ranger session if the request is authenticated.
+	 */
+	@Override
+	protected void doFilter(FilterChain filterChain, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+		String userName = null;
+		boolean checkCookie = response.containsHeader("Set-Cookie");
+		if(checkCookie){
+			Collection<String> authUserName = response.getHeaders("Set-Cookie");
+			if(authUserName != null){
+				Iterator<String> i = authUserName.iterator();
+				while(i.hasNext()){
+					String cookie = i.next();
+					if(!StringUtils.isEmpty(cookie)){
+						if(cookie.toLowerCase().startsWith(AuthenticatedURL.AUTH_COOKIE.toLowerCase()) && cookie.contains("u=")){
+							String[] split = cookie.split(";");
+							if(split != null){
+								for(String s : split){
+									if(!StringUtils.isEmpty(s) && s.toLowerCase().startsWith(AuthenticatedURL.AUTH_COOKIE.toLowerCase())){
+										int ustr = s.indexOf("u=");
+										if(ustr != -1){
+											int andStr = s.indexOf("&", ustr);
+											if(andStr != -1){
+												try{
+													userName = s.substring(ustr+2, andStr);
+												}catch(Exception e){
+													userName = null;
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		// if security context does not have a user session, authenticate the security context and create a session
+		if(!isAuthenticated()){
```

Comment:
lines 170-186: likely can be a small method (self-documented name). And you want to test it (you can add test in future, refactoring will at least give you readability.